client/core/core.go

```diff
@@ -413,29 +423,58 @@ func (dc *dexConnection) processNomatch(oid order.OrderID) error {
 }
 
 // runMatches runs the sorted matches returned from parseMatches.
-func (dc *dexConnection) runMatches(matches map[order.OrderID]*serverMatches) error {
-	for _, match := range matches {
-		tracker := match.tracker
-		if len(match.msgMatches) > 0 {
-			err := tracker.negotiate(match.msgMatches)
+func (dc *dexConnection) runMatches(tradeMatches map[order.OrderID]*serverMatches) error {
+	runMatch := func(sm *serverMatches) error {
+		tracker := sm.tracker
+		errs := newErrorSet("trade %v - ", sm.tracker.ID())
+		// Begin negotiation for any msgMatches.
+		if len(sm.msgMatches) > 0 {
+			err := tracker.negotiate(sm.msgMatches)
 			if err != nil {
-				return err
-			}
-			_, err = tracker.tick()
-			if err != nil {
-				return err
```

Comment:
One tick failure was killing the other matches in https://github.com/decred/dcrdex/issues/554 also.